pkg/cmd/experimental/bundlesecret/bundle-secret.go

```diff
@@ -0,0 +1,114 @@
+
+//export OPENSHIFTCONFIG=./openshift.local.certificates/admin/.kubeconfig
+import (
+	"fmt"
+	kapi "github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	kcmdutil "github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/cmd/util"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util"
+	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
+	"github.com/spf13/cobra"
+	"io"
+	"io/ioutil"
+	"os"
+	"strings"
+)
+
+const MaxSecretSize = 1 * 1024 * 1024
+
+func NewCmdBundleSecret(f *clientcmd.Factory, parentName, name string, out io.Writer) *cobra.Command {
+	var sourcelist util.StringList
+	cmd := &cobra.Command{
+		Use:   "bundle-secret NAME -f SOURCENAME",
+		Short: "bundle data from a source list to load kubernetes secret",
+		Long:  "bundle data from a source list to load kubernetes secret",
+		Run: func(cmd *cobra.Command, args []string) {
+			var err error
+			if len(args) == 0 {
+				err = fmt.Errorf("Must specify NAME as an argument")
+				checkErr(err, out)
+			}
+			objname := args[0]
+			if len(sourcelist) == 0 {
+				err = fmt.Errorf("Must specify files and/or directories to gather data from.")
+				checkErr(err, out)
+			}
+
+			secretdata := make(map[string][]byte)
+			for _, sourceitem := range sourcelist {
+				sourceitem = strings.TrimSuffix(sourceitem, "/")
+				info, err := os.Stat(sourceitem)
+				checkErr(err, out)
+
+				if info.IsDir() {
+					filelist, err := ioutil.ReadDir(sourceitem)
+					checkErr(err, out)
+					for _, item := range filelist {
+						if item.IsDir() {
+							return
+						} else {
+							file := fmt.Sprint(sourceitem, "/", item.Name())
+
+							readValidFiles(file, item.Name(), secretdata, out)
+						}
+					}
+				} else {
+					file := sourceitem
+					readValidFiles(file, info.Name(), secretdata, out)
+				}
+			}
+			secretobj := kapi.Secret{
+				ObjectMeta: kapi.ObjectMeta{Name: objname},
+				Data:       secretdata,
+			}
+			totalsize := 0
+			for _, bytes := range secretdata {
+				totalsize += len(bytes)
+			}
+			if len(secretobj.Data) == 0 {
+				err = fmt.Errorf("No data added to secret.  No files read.")
+				checkErr(err, out)
+			}
+			if totalsize > MaxSecretSize {
+				err = fmt.Errorf("Secret exceeds max size")
+				checkErr(err, out)
+			}
+			setDefaultPrinter(cmd)
+			err = f.Factory.PrintObject(cmd, &secretobj, out)
+			if err != nil {
+				fmt.Fprintf(out, "Error printing Kubernetes Secret object.\n")
+			}
+			checkErr(err, out)
+		},
+	}
+	cmd.Flags().VarP(&sourcelist, "source", "f", "List of filenames, directories to use as source of Kubernetes Secret.Data")
+	kcmdutil.AddPrinterFlags(cmd)
+	return cmd
+}
+func setDefaultPrinter(c *cobra.Command) {
+	flag := c.Flags().Lookup("output")
+	if len(flag.Value.String()) == 0 {
+		flag.Value.Set("json")
+	}
+}
+func checkErr(err error, out io.Writer) {
+	if err == nil {
+		return
+	}
+	fmt.Fprintf(out, "Error: %v\n", err)
+	return
+}
+func readValidFiles(filename string, isValid string, datamap map[string][]byte, out io.Writer) {
```

Comment:
just have this method return an error, and handle the error from the calling function